### PR TITLE
[opentelemtry-collector] Fixed the annotations section for StatefulSet / Deployment

### PR DESCRIPTION
**Pull request by sherifkayad**
`annotations` on the metadata level of Deployments / StatefulSets were not working properly. That PR attempts to fix that.

### CODE REVIEW

**Comment by sherifkayad** [comment]
@TylerHelmuth am not really sure why the linting of the demo chart fails .. tried bumping the version and upgrading the dependency didn't work .. then tried downgrading the dependency also didn't work .. any ideas?

**Comment by sherifkayad** [comment]
@TylerHelmuth I will revert all changes done to the demo chart .. once you can review would be cool to have a look at the CI job and let me know what might be wrong there

**Comment by TylerHelmuth** [comment]
@sherifkayad it looks like you've deleted the demo example.  Make sure your branch has the latest from main and then run "make generate-examples". Only the collector examples should change.

**Comment by sherifkayad** [comment]
@TylerHelmuth I managed to fix the checks. Would be great if you can have another look

**Review by Allex1 — APPROVED** [review]

File: charts/opentelemetry-collector/templates/_helpers.tpl [hunk 1]
```diff
@@ -139,13 +139,6 @@ Return if ingress is stable.
 {{- end }}
 {{- end }}
 
-{{- define "opentelemetry-collector.annotations" -}}
-{{- if and (eq .Values.mode "deployment") .Values.annotations }}
```
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/templates/_helpers.tpl:143]: Instead of removing this helper, can we at statefulset to the condition?

Comment (dmitryax) [reply charts/opentelemetry-collector/templates/_helpers.tpl:143]: I actually agree @sherifkayad. I don't think why the helper is needed anymore. @TylerHelmuth don't we want to apply annotation to daemonset?

Comment (TylerHelmuth) [reply charts/opentelemetry-collector/templates/_helpers.tpl:143]: I don't have a good reason other than at some point we decided to exclude them for daemonsets. I can't think of a good reason to do that so you're both right, removing this is a good idea.

**Review by TylerHelmuth — APPROVED** [review]

File: charts/opentelemetry-collector/templates/_helpers.tpl (hunk 1 above)
Comment (sherifkayad) [reply charts/opentelemetry-collector/templates/_helpers.tpl:143]: @dmitryax @TylerHelmuth thanks guys for the feedback. So I understand we can get this one merged already?

**Review by dmitryax — APPROVED** [review]

**Pull request merged**